core/state_processor.go

```diff
@@ -664,6 +664,13 @@ func (p *StateProcessor) Apply(batch ethdb.Batch, block *types.Block, newInbound
 		return nil, errors.New("failed to load etx set")
 	}
 	time2 := common.PrettyDuration(time.Since(start))
+
+	if p.hc.IsGenesisHash(block.ParentHash(nodeCtx)) {
```

Comment:
This should go in AddGenesisUtxos() function where the alloc accounts are added, not here